### ISSUE TITLE
Training without Sequence Parallelism but VIDEO_SYNC_GROUP

### ISSUE THREAD

**Issue by rob-hen** [jy0205/Pyramid-Flow#162]
Hi all,

the provided script `train_pyramid_flow.sh` does not set the flag `use_sequence_parallel`. In that case, what is the purpose of using `VIDEO_SYNC_GROUP=8`?  Why we want all workers to use the same video?

**Comment by rob-hen** [comment]
I found that if not activated before, in the  attention a sequence parallel group is created, e.g. in `pyramid_dit/flux_modules/modeling_flux_block.py` , function `SequenceParallelVarlenFlashSelfAttentionWithT5Mask`.

**Comment by jy0205** [comment]
Hi, we do not use the sequence parallel during training. The `VIDEO_SYNC_GROUP` controls the number of processes that accept the same video batch as input. We find such a trick will make the gradient direction more stable (optimize the performance of the whole latent sequence of a video, not just a latent from different videos).

**Comment by rob-hen** [comment]
Hi @jy0205,

thank you the answer.
So with `VIDEO_SYNC_GROUP =8` and `GPUS=8`, all GPUs get exactly the same batch.  However, I don't see the difference between the processes, all will use exactly the same latent: https://github.com/jy0205/Pyramid-Flow/blob/e4b02ef31edba13e509896388b1fedd502ea767c/dataset/dataset_cls.py#L192 .

**Comment by yjhong89** [comment]
I think video_sync_group doesn't split same video latent, but accept same video latent without splitting.
- This part is different to sequece parallel, which split latent according to time axis.
- Is that right ??

**Comment by yjhong89** [comment]
- Why only the number of high resolution units are uniformly sampled??
- https://github.com/jy0205/Pyramid-Flow/blob/e4b02ef31edba13e509896388b1fedd502ea767c/pyramid_dit/pyramid_dit_for_video_gen_pipeline.py#L360

**Comment by jy0205** [comment]
> I think video_sync_group doesn't split same video latent, but accept same video latent without splitting.
> 
> * This part is different to sequece parallel, which split latent according to time axis.
> * Is that right ??

Yes, you are right. The `video_sync_group ` does not split the video. It works since different video ranks load different video lengths. You can find in the `sample_length` method.

**Comment by jy0205** [comment]
> * Why only the number of high resolution units are uniformly sampled??
> * https://github.com/jy0205/Pyramid-Flow/blob/e4b02ef31edba13e509896388b1fedd502ea767c/pyramid_dit/pyramid_dit_for_video_gen_pipeline.py#L360
All the stages employ the uniform sampling. We will make the video token sequence length-balanced (let the token length sum to be fixed)
